src/aggregation/metric/extended_stats.rs

```diff
@@ -0,0 +1,1177 @@
+    pub min: Option<f64>,
+    /// The max value of the fast field values.
+    pub max: Option<f64>,
+    /// The average of the fast field values. `None` if count equals zero.
+    pub avg: Option<f64>,
+    /// The sum of squares of the fast field values. `None` if count equals zero.
+    pub sum_of_squares: Option<f64>,
+    /// The variance of the fast field values. `None` if count is less then 2.
+    pub variance: Option<f64>,
+    /// The variance population of the fast field values, always equal to variance. `None` if count
+    /// is less then 2.
+    pub variance_population: Option<f64>,
+    /// The variance sampling of the fast field values, always equal to variance. `None` if count
+    /// is less then 2.
+    pub variance_sampling: Option<f64>,
+    /// The standard deviation of the fast field values. `None` if count is less then 2.
+    pub std_deviation: Option<f64>,
+    /// The standard deviation of the fast field values, always equal to variance. `None` if count
+    /// is less then 2.
+    pub std_deviation_population: Option<f64>,
+    /// The standard deviation sampling of the fast field values. `None`
+    /// if count is less then 2.
+    pub std_deviation_sampling: Option<f64>,
+    /// The standard deviation bounds of the fast field values, always equal to variance. `None`
+    /// if count is less then 2.
+    pub std_deviation_bounds: Option<StandardDeviationBounds>,
+}
+
+/// A sub struct for ExtendedStat containing deviation bounds
+/// the values depend on sigma and represent
+/// the bounds from the average with a distance of
+/// std_deviation*sigma
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct StandardDeviationBounds {
+    /// upper bound -> avg + std_dev*sigma
+    pub upper: f64,
+    /// lower bound -> avg - std_dev*sigma
+    pub lower: f64,
+    /// upper bound sampling -> avg + std_dev_sampling*sigma
+    pub upper_sampling: f64,
+    /// lower bound sampling -> avg - std_dev_sampling*sigma
+    pub lower_sampling: f64,
+    /// same as upper
+    pub upper_population: f64,
+    /// same as lower
+    pub lower_population: f64,
+}
+
+impl ExtendedStats {
+    pub(crate) fn get_value(&self, agg_property: &str) -> crate::Result<Option<f64>> {
+        match agg_property {
+            "count" => Ok(Some(self.count as f64)),
+            "sum" => Ok(Some(self.sum)),
+            "min" => Ok(self.min),
+            "max" => Ok(self.max),
+            "avg" => Ok(self.avg),
+            "variance" => Ok(self.variance),
+            "variance_sampling" => Ok(self.variance_sampling),
+            "variance_population" => Ok(self.variance_population),
+            "sum_of_squares" => Ok(self.sum_of_squares),
+            "std_deviation" => Ok(self.std_deviation),
+            "std_deviation_sampling" => Ok(self.std_deviation_sampling),
+            "std_deviation_population" => Ok(self.std_deviation_population),
+            "std_deviation_bounds.lower" => Ok(self
```

Comment:
this should match the elastic search name, not the internal name of the property

Reply:
Perhaps I'm missing something, but the names seem to match the ElasticSearch ones https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-extendedstats-aggregation.html